service_discovery/samples/simple-discovery/provider/src/main.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+//! A simple provider for a sample of Chariott Service Discovery.
+//!
+//! This provider has one service, the hello_world service, which has one
+//! method that returns a message containing "Hello, " followed by the string
+//! provided in the request. The provider registers itself with Chariott.
+
+// Tells cargo to warn if a doc comment is missing and should be provided.
+#![warn(missing_docs)]
+
+use chariott_common::error::Error;
+use hello_world_impl::HelloWorldImpl;
+use proto_servicediscovery::hello_world::v1::hello_world_server::HelloWorldServer;
+use std::net::SocketAddr;
+use url::Url;
+
+use proto_servicediscovery::chariott_registry::v1::registry_client::RegistryClient;
+use proto_servicediscovery::chariott_registry::v1::{RegisterRequest, ServiceMetadata};
+use tonic::transport::Server;
+use tonic::Request;
+use tracing::info;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::EnvFilter;
+
+mod hello_world_impl;
+
+/// URL for the chariott service registry
+const CHARIOTT_SERVICE_REGISTRY_URL: &str = "http://0.0.0.0:50000";
```

Comment:
Even though this is a sample, you may want to consider creating a config file for these value.

Reply:
I have created this issue to add configuration in a follow-up change: https://github.com/eclipse-chariott/chariott/issues/144. I will think about if we want configs for samples as well.